Issue 892

RyanDJLee opened the pull request:
- Fix implementation of Show for From and Progreq.

david-yz-liu commented:
@RyanDJLee I wonder if the reason the compilation error wasn't detected before is that this module isn't listed in `courseography.cabal`. You should add it there and try to run a "stack build".

The module name should be changed to `Database.Requirements` (at the top of the file), and added to the cabal file.

david-yz-liu commented on app/Database/Requirement.hs:
```diff
@@ -12,28 +12,28 @@ module Requirement
+import qualified Data.List as L
 
-
-data ProgramReq = PRGREQ String [Req]
```
I'm not sure why you changed this; it seems to be the case that a program will have multiple requirements. I guess you could represent this as an implicit "AND", but it will be useful to keep them separate at the top level, I think.

RyanDJLee commented:
Changed it back! It does make a lot more sense to keep program requirements as a list of requirements.

RyanDJLee commented:
- Changed value constructor of program requirements back to list of reqs.
- Changed name of Requirement module and added it to list the of dependencies.
- Build worked well and tested with stack repl.

david-yz-liu commented:
Thanks, @RyanDJLee!
